### ISSUE TITLE
nan return by nn.CrossEntropyLoss when all the labels are ignore_index in torch 1.11

### ISSUE THREAD

**Issue by gitabtion** [pytorch/pytorch#70348]
### 🐛 Describe the bug

If you update torch to 1.11, the following issue will be occur:

```python
import torch
loss_fct = torch.nn.CrossEntropyLoss()

loss = loss_fct(torch.rand(10,5), torch.tensor([-100]*5))
print(loss)
# tensor(nan)
```

### Versions

docker image: nvcr.io/nvidia/pytorch:21.12-py3
torch version: 1.11.0a0+b6df043

**Comment by jbschlosser** [comment]
Hey @gitabtion, thanks for the report! This was done intentionally in #64572 to fit better with the mathematical definition of taking the mean of an empty set (see discussion there for more details).

For your purposes, do you expect the old behavior of returning 0.0 when all labels == `ignore_index`? cc @Lezcano

**Comment by gitabtion** [comment]
@jbschlosser Yes, I think returning 0.0 maybe better. Even though it can backward now, but it becomes much inconvenient from this update.

**Comment by Lezcano** [comment]
If you pass an empty tensor to this function it'll return NaNs, since this operation does not have an identity. Why should this case behave differently?

**Comment by mfoglio** [comment]
Hi @jbschlosser , I agree with @gitabtion , it would be nice if the loss returned 0 when all the labels are equal to the specified `ignore_index`. For instance, consider a classifier with 2 tails: one predict the shape of a polygon, the other one predict the color of the polygon. Both the color and the shape have 5 possible values. Suppose that the dataset only rarely contains the information for the shape of a polygon. The loss associated to the shape will often be nan. When summing the two losses in order to backpropagate we will often end up with a nan loss. Of course we could just avoid including the nan loss values into the sum, but returning 0 would be probably cleaner.
